validator/client/propose.go

```diff
@@ -203,3 +234,42 @@ func (v *validator) signBlock(ctx context.Context, pubKey [48]byte, epoch uint64
 	}
 	return sig.Marshal(), nil
 }
+
+// Sign voluntary exit with proposer domain and private key.
+func (v *validator) signVoluntaryExit(ctx context.Context, pubKey [48]byte, exit *ethpb.VoluntaryExit) ([]byte, error) {
+	domain, err := v.domainData(ctx, exit.Epoch, params.BeaconConfig().DomainVoluntaryExit[:])
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get domain data")
+	}
+	var sig bls.Signature
+
+	exitRoot, err := helpers.ComputeSigningRoot(exit, domain.SignatureDomain)
```

Comment:
can domain ever be nil?

Reply:
Wouldn't hurt to check this since it's coming from an RPC call.